src/bin/drag_and_drop_listbox.rs

```diff
@@ -0,0 +1,193 @@
+// A drag-and-drop example. You can drag the list items using the icon on the left, and drop them
+// to other icons to move the list item. Originally listbox-dnd.c in GTK+ repository. Ported to
+// gtk-rs by Ömer Sinan Ağacan <omeragacan@gmail.com>.
+
+extern crate cairo;
+extern crate gdk;
+extern crate gio;
+extern crate gtk;
+
+use gdk::{Atom, DragAction, DragContext, ModifierType, Screen};
+use gio::prelude::*;
+use gtk::*;
+use std::env::args;
+
+fn main() {
+    let app = gtk::Application::new("com.github.gtk-rs.examples.basic", Default::default())
+        .expect("Initialization failed...");
+
+    app.connect_activate(|app| {
+        build_ui(app);
+    });
+
+    app.run(&args().collect::<Vec<_>>());
+}
+
+fn build_ui(app: &gtk::Application) {
+    let css_provider = CssProvider::new();
+    css_provider.load_from_data(CSS.as_bytes()).unwrap();
+    StyleContext::add_provider_for_screen(
+        Screen::get_default().as_ref().unwrap(),
+        &css_provider,
+        STYLE_PROVIDER_PRIORITY_APPLICATION,
+    );
+
+    let window = gtk::ApplicationWindow::new(app);
+    window.set_title("A drag-and-drop example");
+    window.set_default_size(350, 500);
+
+    let list_box = ListBox::new();
+    list_box.set_selection_mode(SelectionMode::None);
```

Comment:
Shouldn't this be `::Single` so that a single item can be selected (and dragged)?

Reply:
That might be better from UX point of view, but the original example that I ported did exactly this.

Reply:
Fine with me :) Does it change anything with the behaviour of the application?

Reply:
> Does it change anything with the behaviour of the application?

No functional changes but it looks strange because you can't do anything with the selected items.

Reply:
Ack, let's just keep it as is then :)